ambry-router/src/main/java/com.github.ambry.router/NonBlockingRouter.java

```diff
@@ -469,11 +475,22 @@ private void onResponse(List<ResponseInfo> responseInfoList) {
      */
     @Override
     public void run() {
+      long startTime;
+      long lastTime;
       try {
         while (isOpen.get()) {
+          startTime = time.milliseconds();
+          lastTime = startTime;
           List<RequestInfo> requestInfoList = pollForRequests();
+          routerMetrics.operationControllerPollforRequestTimeMs.update(time.milliseconds() - lastTime);
+          lastTime = time.milliseconds();
           List<ResponseInfo> responseInfoList = networkClient.sendAndPoll(requestInfoList);
+          routerMetrics.numOfRequestInfosPerPoll.update(requestInfoList.size());
```

Comment:
as discussed, since `sendAndPoll` is non-blocking and there are limited number of connections, the time it takes should not be dependent on the number of requests. The time this method takes will be more or less bounded by the poll timeout. If it takes longer, we will have to find out why. Also note that `sendAndPoll` can possibly be called with zero requests as well - that is, it can be called just for the purposes of polling.